TM1py/Services/GitService.py

```diff
@@ -0,0 +1,179 @@
+        :param Branch: The name of source branch
+        :param Force: A flag passed in for evaluating preconditions
+        :param Username: GIT username
+        :param Password: GIT password
+        :param PublicKey: SSH public key, available from PAA V2.0.0.4
+        :param PrivateKey: SSH private key, available from PAA V2.0.0.4
+        :param Passphrase: Passphrase for decrypting private key, if set
+        """
+        url = "/api/v1/GitPull"
+        body = json.dumps(kwargs)
+        response = self._rest.POST(url=url, data=body)
+        if execute:
+            planid = json.loads(response.content).get('ID')
+            self.git_executeplan(planid=planid)
+
+        return response
+
+    def git_executeplan(self, planid) -> Response:
+        """ Executes a plan based on the planid
+        :param planid: GitPlan id
+        """
+
+        url = "/api/v1/GitPlans('{}')/tm1.Execute".format(planid)
+        return self._rest.POST(url=url)
+
+    def git_getplans(self, **kwargs) -> List[GitPlan]:
+        """ Gets a list of currently available GIT plans
+        """
+        url = "/api/v1/GitPlans"
+        plans = []
+        response_dict = json.loads(self._rest.GET(url=url, body=kwargs).content)
+        # Every individual plan is wrapped in a "value" parent, iterate through those to get the actual plans
+        for plan in response_dict.get('value'):
+            plan_id = plan.get('ID')
+            # Check if plan has an ID, sometimes there's a null in the mix that we don't want
+            if plan_id is not None:
+                plan_branch = plan.get('Branch')
+                plan_force = plan.get('Force')
+
+                # A git plan can either be a PushPlan or a PullPlan, these have slightly different variables, so we need to handle those differently
+                if plan.get('@odata.type') == '#ibm.tm1.api.v1.GitPushPlan':
+                    plan_new_branch = plan.get('NewBranch')
+                    plan_source_files = plan.get('SourceFiles')
+
+                    new_commit = GitCommit(
+                                commit_id=plan.get('NewCommit').get('ID'),
+                                summary=plan.get('NewCommit').get('Summary'),
+                                author=plan.get('NewCommit').get('Author'))
+
+                    parent_commit = GitCommit(
+                                commit_id=plan.get('ParentCommit').get('ID'),
+                                summary=plan.get('ParentCommit').get('Summary'),
+                                author=plan.get('ParentCommit').get('Author'))
+
+                    current_plan = GitPushPlan(
+                                plan_id=plan_id, branch=plan_branch, force=plan_force,
+                                new_branch=plan_new_branch, new_commit=new_commit,
+                                parent_commit=parent_commit, source_files=plan_source_files)
+
+                if plan.get('@odata.type') == '#ibm.tm1.api.v1.GitPullPlan':
```

Comment:
I think it would be good to use `elif` here and ideally include an `else` clause that throws a `ValueError` indicating that the plan is illegal